android-core/src/androidTest/kotlin/com.mparticle/networking/MParticleBaseClientImplTest.kt

```diff
@@ -20,11 +23,166 @@ class MParticleBaseClientImplTest : BaseCleanInstallEachTest() {
+            Pair("eu1-bkabfno0b8zpv5bwi2zm2mfa1kfml19al", "eu1"),
+            Pair("au1-iermuj83dbeoshm0n32f10feotclq6i4a", "au1"),
+            Pair("st1-k77ivhkbbqf4ce0s3y12zpcthyn1ixfyu", "st1"),
+            Pair("us3-w1y2y8yj8q58d5bx9u2dvtxzl4cpa7cuf", "us3"),
+            Pair("kajsdhasdiuyaiudiashhadjhdasjk", "us1")
+        )
+        val identityUrl = mRandomUtils.getAlphaString(20)
+        val configUrl = mRandomUtils.getAlphaString(20)
+        val audienceUrl = mRandomUtils.getAlphaString(20)
+        val eventsUrl = mRandomUtils.getAlphaString(20)
+        val options = MParticleOptions.builder(mContext)
+            .credentials(apiKey, "secret")
+            .networkOptions(
+                NetworkOptions.builder()
+                    .addDomainMapping(
+                        DomainMapping.audienceMapping(audienceUrl)
+                            .build()
+                    )
+                    .addDomainMapping(
+                        DomainMapping.configMapping(configUrl)
+                            .build()
+                    )
+                    .addDomainMapping(
+                        DomainMapping.identityMapping(identityUrl)
+                            .build()
+                    )
+                    .addDomainMapping(DomainMapping.eventsMapping(eventsUrl).build())
+                    .build()
+            )
+            .build()
+        MParticle.start(options)
+        val baseClientImpl = AccessUtils.getApiClient() as MParticleBaseClientImpl
+        map.forEach { key, value ->
+            mConfigManager?.setCredentials(key, value)
+            val prefix = mConfigManager?.podPrefix
+            assertEquals(value, prefix)
+            assertEquals(
+                "${NetworkOptionsManager.MP_URL_PREFIX}.$prefix.mparticle.com",
+                baseClientImpl.getPodUrl(NetworkOptionsManager.MP_URL_PREFIX, prefix, true)
+            )
+            assertEquals(
+                "${NetworkOptionsManager.MP_IDENTITY_URL_PREFIX}.$prefix.mparticle.com",
+                baseClientImpl.getPodUrl(
+                    NetworkOptionsManager.MP_IDENTITY_URL_PREFIX,
+                    prefix,
+                    true
+                )
+            )
+
+            assertEquals(
+                "${NetworkOptionsManager.MP_URL_PREFIX}.mparticle.com",
+                baseClientImpl.getPodUrl(
+                    NetworkOptionsManager.MP_URL_PREFIX,
+                    prefix,
+                    false
+                )
+            )
+            assertEquals(
+                "${NetworkOptionsManager.MP_IDENTITY_URL_PREFIX}.mparticle.com",
+                baseClientImpl.getPodUrl(
+                    NetworkOptionsManager.MP_IDENTITY_URL_PREFIX,
+                    prefix,
+                    false
+                )
+            )
+        }
+    }
+
     @Test
     @Throws(MalformedURLException::class)
     fun testGetUrlForceDefaultOption() {
```

Comment:
This test would be removed in the future since we are trying to get rid of that `forceDefaultUrl` boolean since that was only for the recursion issue, right?

Reply:
yes agree in future will remove this test cases